apis/v1alpha1/validation_test.go

```diff
@@ -89,6 +91,81 @@ func TestValidateResouceMapping(t *testing.T) {
 				Resource: &Resource{
 					Provider: "gcp",
 					Name:     "//pubsub.googleapis.com/projects/test-project/topics/test-topic",
+					Subscope: "parent/foo/child/bar?key1=value1&key2=value2",
```

Comment:
add tc.wantSubscope to verify the logic works as expected. Same for other test cases when needed.

Reply:
Added.